src/vector.carp

```diff
@@ -0,0 +1,150 @@
+
+(load "https://github.com/scolsen/typeclass@master")
+(load "src/bound.carp")
+
+(doc Vector "
+  Vector is an array-like type that supports length/bounds checking at
+  type-level. It also implements the [dimensional interfaces](dimensional.html)
+  and supports generic programming.")
+(deftype (Vector s a) [elements (Array a)])
+(private Vector.elements)
```

Comment:
Should we make the setters private as well?

Reply:
Good call!